src/uct/ib/rc/base/rc_ep.h

```diff
@@ -365,7 +364,9 @@ uct_rc_txqp_posted(uct_rc_txqp_t *txqp, uct_rc_iface_t *iface, uint16_t res_coun
         --iface->tx.cq_available;
         UCS_STATS_UPDATE_COUNTER(txqp->stats, UCT_RC_TXQP_STAT_SIGNAL, 1);
     } else {
-        txqp->unsignaled++;
+        ++txqp->unsignaled;
+        /* reserve to error */
+        --iface->tx.cq_available;
```

Comment:
optimization: can we do something like iface->tx.cq_available -= txqp->unsignaled, only when posting a signaled send?